File: src/AddTwoInts/add_server.py
import rospy
from beginning_with_ros.srv import AddTwoInts
from beginning_with_ros.srv import AddTwoIntsRequest
from beginning_with_ros.srv import AddTwoIntsResponse

# Provide response
def add_handler(request):
    response = request.a + request.b
    rospy.loginfo("")
    print('    Request from client: {}, {}'.format(request.a, request.b))
    print('    Response: {}'.format(sum))
    return response

def add_server():
    rospy.init_node('add_server_py')

    # Initialize service provider
    rospy.Service('add_two_ints', AddTwoInts, add_handler)

    rospy.loginfo('Server ready')

    # Wait for request
    rospy.spin()

if __name__ == "__main__":
    add_server()

File: src/AddTwoInts/add_server.cpp
#include "ros/ros.h"
#include "beginning_with_ros/AddTwoInts.h"
#include <stdio.h>

bool add_handler(beginning_with_ros::AddTwoInts::Request &request,
                 beginning_with_ros::AddTwoInts::Response &response)

{
    response.sum = request.a + request.b;

    ROS_INFO("");    
    printf("    Request from client: %ld, %ld\n", (long int)request.a, (long int)request.b);
    printf("    Response: %ld\n", (long int)response.sum);

    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "add_server_cpp");
    ros::NodeHandle nodeHandle;
    ros::ServiceServer server = nodeHandle.advertiseService("add_two_ints", add_handler);
    ROS_INFO("Server ready");
    ros::spin();

    return 0;
}